packages/dev/inspector-v2/src/extensibility/builtInsExtensionFeed.ts

```diff
@@ -1,43 +1,31 @@
 import type { IExtensionFeed, ExtensionMetadata, IExtensionMetadataQuery, ExtensionModule } from "./extensionFeed";
 
-// This file contains "built in" extensions. They are optional and installed/uninstalled by the user, but they are
-// well-known at build time and the extension is "downloaded" by simply doing a dynamic import. This is different
-// from future extension types that are built and published apart from the inspector, and are downloaded as an isolated script.
-
-const CreationToolsExtensionMetadata = {
-    name: "Asset Creation",
-    description: "Adds new features to enable creating Babylon assets such as node materials, flow graphs, and more.",
-    keywords: ["creation"],
-} as const;
-
-const ExportToolsExtensionMetadata = {
-    name: "Export Tools",
-    description: "Adds new features to enable exporting Babylon assets such as .gltf, .glb, .babylon, and more.",
-    keywords: ["export", "gltf", "glb", "babylon", "exporter", "tools"],
-} as const;
-
-const CaptureToolsExtensionMetadata = {
-    name: "Capture Tools",
-    description: "Adds new features to enable capturing screenshots, GIFs, videos, and more.",
-    keywords: ["capture", "screenshot", "gif", "video", "tools"],
-} as const;
-
-const ImportToolsExtensionMetadata = {
-    name: "Import Tools",
-    description: "Adds new features related to importing Babylon assets.",
-    keywords: ["import", "tools"],
-} as const;
-
-const Extensions: readonly ExtensionMetadata[] = [/*CreationToolsExtensionMetadata, */ ExportToolsExtensionMetadata, CaptureToolsExtensionMetadata, ImportToolsExtensionMetadata];
+export type BuiltInExtension = ExtensionMetadata & {
+    /**
+     * Gets the extension module, typically dynamically importing the extension.
+     * @returns The extension module (e.g. a collection of ServiceDefinitions).
+     */
+    getExtensionModuleAsync(): Promise<ExtensionModule>;
+};
 
 /**
- * @internal
+ * A simple extension feed implementation that provides a fixed set of "built in" extensions.
+ * "Built in" in this context means extensions that are known at bundling time, and included
+ * in the bundle. Each extension can be dynamically imported so they are split into separate
+ * bundle chunks and downloaded only when first installed.
  */
 export class BuiltInsExtensionFeed implements IExtensionFeed {
-    public readonly name = "Built-ins";
+    private readonly _extensions: readonly BuiltInExtension[];
+
+    public constructor(
+        public readonly name: string,
+        extensions: Iterable<BuiltInExtension>
+    ) {
+        this._extensions = Array.from(extensions);

```

Comment:
why creating a new array ? is it to protect the one passed in ?

Reply:
I was just making the inputs flexibly, but storing as an array to get the usual array functions like filtering. I'm not super attached though, I could just switch the input to be more specifically an array, and if you are starting with something else for some reason (like a Set), then you'd just have to convert to an array yourself. I'm ok either way if you have a strong preference.